noxfile.py

```diff
@@ -373,6 +377,7 @@ def test_core(session):
 @nox.session(python=PYTHON_VERSIONS)
 def test_plugins(session):
     _upgrade_basic(session)
+    _install_pytest(session)
```

Comment:
nit: since we need to always install pytest along with the upgrade_basic, could we merge them into one method, something like install_basic?

Reply:
Actually the `lint` and `lint_plugins` nox sessions do not require install pytest (but they do still use upgrade_basic).

Reply:
I have just opened PR #2016 which only pins pytest for the jupyter notebook session. I'm thinking of closing this PR in favor of that one. What do you think?

(One benefit of not pinning pytest is that the new pytest version should allow us to move forward with [python3.10 support](https://github.com/facebookresearch/hydra/issues/1856), which was previously blocked).